### PR TITLE
[mono] Turn methods with no instance access to static

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

**Comment by CoffeeFlux** [comment]
Are there any plans for a CoreCLR equivalent to this? Looking at things like `AppendStackFrameWithoutMethodBase`, the same change could be applied there as well.

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs [hunk 1]
```diff
@@ -419,13 +419,13 @@ public void SetCustomAttribute(ConstructorInfo con, byte[] binaryAttribute)
...
 
-        private Exception not_supported()
+        private static Exception not_supported()
         {
             // Strange message but this is what MS.NET prints...
             return new NotSupportedException("The invoked member is not supported in a dynamic module.");
         }
 
-        private string create_assembly_version(string version)
+        private static string create_assembly_version(string version)
```
Comment (CoffeeFlux) [inline src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs:428]: This seems to be unused, I think we can just get rid of it?
```
ryan@kenshin:~/git/runtime$ git grep create_assembly_version
src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs:        private string create_assembly_version(string version)
```

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs [hunk 2]
```diff
@@ -467,13 +467,13 @@ private string create_assembly_version(string version)
             return ver[0] + "." + ver[1] + "." + ver[2] + "." + ver[3];
         }
 
-        private string GetCultureString(string str)
+        private static string GetCultureString(string str)
```
Comment (CoffeeFlux) [inline src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs:470]: Unused?
```
ryan@kenshin:~/git/runtime$ git grep GetCultureString
src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs:        private string GetCultureString(string str)
```

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs [hunk 3]
```diff
@@ -947,7 +948,9 @@ public virtual void EndExceptionBlock()
             InternalEndClause();
             MarkLabel(ex_handlers[cur_block].end);
             ex_handlers[cur_block].End(code_len);
+#if FALSE
```
Comment (CoffeeFlux) [inline src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs:951]: Isn't the whole point of the way the defines are set up now to make it easy to turn these prints on? This makes it more annoying; is there another option here?

File: src/mono/netcore/System.Private.CoreLib/src/System/RuntimeType.Mono.cs [hunk 4]
```diff
@@ -422,14 +422,14 @@ private static bool FilterApplyPrefixLookup(MemberInfo memberInfo, string? name,
             RuntimeMethodInfo method, BindingFlags bindingFlags, CallingConventions callConv, Type[]? argumentTypes)
         {
             // Optimization: Pre-Calculate the method binding flags to avoid casting.
-            return FilterApplyMethodBase(method, method.BindingFlags, bindingFlags, callConv, argumentTypes);
+            return FilterApplyMethodBase(method, 0, bindingFlags, callConv, argumentTypes);
```
Comment (CoffeeFlux) [inline src/mono/netcore/System.Private.CoreLib/src/System/RuntimeType.Mono.cs:425]: If this argument is unused we should just get rid of it. This method is internal to our RuntimeType implementation.

File: src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs [hunk 5]
```diff
@@ -534,6 +534,7 @@ private void SetGenericValueImpl<T>(int pos, ref T value)
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         private extern void SetValueRelaxedImpl(object? value, int pos);
 
+#pragma warning disable CA1822
```
Comment (CoffeeFlux) [inline src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs:537]: Why are these different? Is it just the volume and number of callers?

Comment (am11) [reply src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs:537]: I think the methods which are accessing `Length` property can be excluded from this suppression?

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs (hunk 1 above)
Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs]: Removed

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs (hunk 3 above)
Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs]: Removed

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs (hunk 2 above)
Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.Mono.cs]: Removed

File: src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs (hunk 5 above)
Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs:537]: They are accessed as instance calls from JIT

Comment (am11) [reply src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs:537]: Looks like `InternalArray__ICollection_get_Count`, `InternalArray__IReadOnlyCollection_get_Count` and few others which are accessing class members do not need the suppression.

**Review by CoffeeFlux — APPROVED** [review]
I think [this comment](https://github.com/dotnet/runtime/pull/43228#discussion_r502791444) is worth addressing, but otherwise looks good to me. Thanks!

File: src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs (hunk 5 above)
Comment (marek-safar) [reply src/mono/netcore/System.Private.CoreLib/src/System/Array.Mono.cs:537]: The intention is to indicate that none of the methods should be made static even if they could

**Pull request merged**